Makes read_vars() skip unreadable variables. Fixes Debian bug 768880.

Old versions of efibootmgr and libefivar0 allowed the installation of
necessary EFI boot entries on the affected systems. It seems that an old
libefivar0 did not report read errors.

Then libefivar0 started to crash on unreadable variables (see Debian bug
764386). Since that bug has been fixed, efibootmgr doesn't crash anymore,
but the read errors are reported and efibootmgr always fails with an
input/output error.

Signed-off-by: Jan Echternach <jan@goneko.de>
Signed-off-by: Daniel Jared Dominguez <Jared_Dominguez@Dell.com>

--- a/src/efibootmgr/efibootmgr.c
+++ b/src/efibootmgr/efibootmgr.c
@@ -110,8 +110,12 @@
 			rc = efi_get_variable(EFI_GLOBAL_GUID, namelist[i],
 					       &entry->data, &entry->data_size,
 					       &entry->attributes);
-			if (rc < 0)
-				goto err;
+                       if (rc < 0) {
+				/* skip unreadable variable */
+				warn("Skipping unreadable variable %s", namelist[i]);
+				free(entry);
+				continue;
+			}
 
 			/* latest apple firmware sets high bit which appears
 			 * invalid to the linux kernel if we write it back so
